packages/grid/x-data-grid/src/hooks/core/useGridLocaleText.tsx

```diff
@@ -10,10 +10,10 @@ export const useGridLocaleText = (
 ): void => {
   const getLocaleText = React.useCallback(
```

Comment:
This seems to work fine

```suggestion
  const getLocaleText = React.useCallback<GridLocaleTextApi['getLocaleText']>(
    (key) => {
      if (props.localeText[key] == null) {
        throw new Error(`Missing translation for key ${key}.`);
      }
      return props.localeText[key];
    },
    [props.localeText],
  );
```